environment/stats/src/cloud/benchflow/collectors/stats.go

```diff
@@ -110,34 +131,16 @@ func stopCollecting(w http.ResponseWriter, r *http.Request) {
 		fmt.Fprintf(w, "Currently not collecting")
 		return
 	}
+	for _, c := range containers {
+		c.stopChannel <- true
+		}
 	collecting = false
 	fmt.Fprintf(w, "Stopped collecting")
```

Comment:
This is not enough, you need a waiting group to be sure about the time the goroutine effectively ends: http://adampresley.com/2015/02/16/waiting-for-goroutines-to-finish-running-before-exiting.html#waiting-for-them-to-finish